src/legacy/core_plugins/kibana/public/management/sections/index_patterns/edit_index_pattern/create_edit_field/create_edit_field.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { withRouter, RouteComponentProps } from 'react-router-dom';
+// @ts-ignore
+import { FieldEditor } from 'ui/field_editor';
+
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { IndexHeader } from '../index_header';
+import { IndexPattern, IndexPatternField } from '../../../../../../../../../plugins/data/public';
+import { ChromeDocTitle, NotificationsStart } from '../../../../../../../../../core/public';
+
+type CreateEditFieldProps = RouteComponentProps & {
+  indexPattern: IndexPattern;
+  mode?: string;
+  fieldName?: string;
+  fieldFormatEditors: any;
+  getConfig: (name: string) => any;
+  servises: {
+    notifications: NotificationsStart;
+    docTitle: ChromeDocTitle;
+    http: Function;
+  };
+};
+
+export const CreateEditField = withRouter(
+  ({
+    indexPattern,
+    mode,
+    fieldName,
+    fieldFormatEditors,
+    getConfig,
+    servises,
+    history,
+  }: CreateEditFieldProps) => {
+    const field: IndexPatternField | undefined =
+      mode === 'edit' && fieldName
+        ? indexPattern.fields.getByName(fieldName)
+        : new IndexPatternField(indexPattern, {
+            scripted: true,
+            type: 'number',
+          });
+
+    if (mode === 'edit') {
+      if (!field) {
+        const message = i18n.translate('kbn.management.editIndexPattern.scripted.noFieldLabel', {
```

Comment:
I think we should use one pattern like e.g. in your previous PR.
Please move it out of class and rename message -> noFieldLabel

Reply:
Unfortunately, no, because defaultMessage for this message depends on values from props.

Reply:
ok, thank you